### ISSUE TITLE
Allow generating a document as an HTML site (clause = single HTML file)

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-standoc#440]
This is the default behavior of ISO 10303 SMRL.

For Part 41 which exceeds 6MB in a single HTML file, browsers are challenged in just opening it. Scrolling/clicking on internal links around is also slow. 

To accommodate lower resourced devices and larger document collections we must to have a way to build a document (or a document collection) into a paged site.

An attribute like this could work:
```
:generate_html_page_at_level: 0 // => disabled?
:generate_html_page_at_level: 1 // => document level
:generate_html_page_at_level: 2 // => clause level
:generate_html_page_at_level: 3 // => sub-clause level
// etc...
```

**Comment by opoudjis** [comment]
This is going to be involved and iterative, and I am not going to deliver anything near as fancy as what you are requesting in a hurry. You are getting clause level.

**Comment by opoudjis** [comment]
Need to split Presentation XML, not compile separate fragments of Semantic XML; in turn, that means Presentation XML needs to be compiled in metanorma

**Comment by opoudjis** [comment]
Also need to be able to compile a document without prefatory material (boilerplate, coverpage): https://github.com/metanorma/metanorma-standoc/issues/440

**Comment by opoudjis** [comment]
> Need to split Presentation XML, not compile separate fragments of Semantic XML; in turn, that means Presentation XML needs to be compiled in metanorma

... Not sure it does: HTML is always compiled from Presentation XML, Presentation XML is in same folder always as Semantic XML, since both are generated from isodoc...

**Comment by opoudjis** [comment]
The HTML collection is being generated, but the index is not, and neither are the headings...

**Comment by opoudjis** [comment]
Adding "presentation-xml" directive to Metanorma collections: indicates that the XML being passed into the collection is already Presentation XML, and is not to be processed further to generate new Presentation XML.

Adding "bare-after-first" directive to Metanorma: only the first file in the manifest is generated as a complete document (coverpage, boilerplate), the remainder are processed as bare files.

**Comment by opoudjis** [comment]
Bug in metanorma: if two internal references point to the same file, only a single reference was being generated in parsing, because one was overwriting the other.

**Comment by opoudjis** [comment]
We allow the doc identifier in the collection to contain markup. (Section title XML is treated as the doc identifier.)

Debugged i18n index for navigation element.

**Comment by opoudjis** [comment]
svgmap references are being resolved in the unitary Presentation XML compiled, before it is being split up into clauses. That causes problems downstream: the references get broken in the svgmap when it is broken up. if the sectionsplit attribute is present, we will postpone svgmap processing.

**Comment by opoudjis** [comment]
I've ended up instead reconstructing the svgmap when the xrefs are already resolved.

**Comment by opoudjis** [comment]
Doing optimisation of the splitting

**Comment by opoudjis** [comment]
Problem: index of collection which contains attachments and sectionsplit (a) does not show the division of the source document into subdocuments, and (b) is displaying the attachments, which is undesirable: the attachments are linked to inline from the Annex of our target document.

(a) is happening because the collection cover is being generated before the files are processed, and the sectionsplit is being done on its contents.

For (b), introduce `index: false` into manifest file entry

**Comment by opoudjis** [comment]
Run fontist installation only once per collection.

Add bibliographic entry only once per filesplit: if already have hidden ref in place for internal reference, don't generate a new one.

**Comment by opoudjis** [comment]
Down to 10 mins from 59, though there will be errors in there, which I can now start debugging, because the compilation time is reasonable

**Comment by opoudjis** [comment]
collection cover for sectionsplit file was overwriting collection cover for entire collection. Collection was splitting input sectionsplit file twice.

**Comment by opoudjis** [comment]
Almost there, but now links to attachments are broken, and they used to work.

**Comment by opoudjis** [comment]
Fixed.
